lib/datadog/core/configuration/settings.rb

```diff
@@ -611,14 +611,36 @@ def initialize(*_)
           #
           # @default `nil`.
           # @return [String,nil]
-          option :service
+          option :service do |o|
+            o.type :string, nilable: true
+          end
 
           # Declare additional service names to bind to remote configuration. Use when
           # integrations are configured to use names that differ from DD_SERVICE.
           #
           # @default `nil`.
           # @return [Array<String>,nil]
-          option :extra_services
+          option :extra_services do |o|
```

Comment:
Since `extra_services` also effectively includes `service`, should we deprecate `service` and ask people to use `extra_services` instead?

Reply:
The duplication is sorted out backend-side.

Hopefully `extra_services` gets to disappear if we manage to figure a way to automate it, while `service` could still be needed for a while. I would not want the user configuration to break then.

Reply:
To clarify: since this can be tricky it would be in scope for a second PR following this one.

Reply:
Attempted here: https://github.com/DataDog/dd-trace-rb/pull/3168